woocommerce/woocommerce-ios
# [Order form] Extract properties (marked with TODO) from `ProductRowViewModel`

Closes: #11412


## Why
This extracts properties (previously marked with TODO) used in `CollapsibleProductRowCardViewModel` from `ProductRowViewModel`. This is another step toward disentangling the logic used in `CollapsibleProductCard` from `ProductRowViewModel`, eventually allowing the products section in the order form to be displayed using `OrderItem` data.

## How
There were four properties marked with TODO:

- Configurable properties: These were not removed from `ProductRowViewModel` because it uses them in the product selector to configure a bundle before adding it to an order.
   -  `isConfigurable`: This property is used to determine if an order item can be configured (e.g. a product bundle).
   -  `configure`: This property is used to configure a configurable order item.
-  `priceAfterDiscountLabel`: This was an unused property, which was removed. (`CollapsibleProductRowCard` uses the `totalPriceAfterDiscountLabel` property, which is already part of the VM.)
-  `orderProductDetailsLabel`: This property displays details about the order item product or variation (the product type, attributes, and/or stock.)

The last property was moved in 86454d5d836ebd8c5de5eb98fa3ace7e8a8fd038. This label depends on properties of the product/variation related to the order item. Rather than adding all of the product properties it depends on to `CollapsibleProductRowCardViewModel`, I decided to keep the VM simple by only adding the label property (`productDetailsLabel`) and creating the label text when the view model is initialized.

For now, there are no changes to what data is displayed in that label — this PR only extracts the properties as-is to the new VM — but in #11389 we can update it to get the attributes directly from the `OrderItem` instead of the `ProductVariation`. (It probably makes sense to add convenience initializers that take a `Product` or `ProductVariation`, as well as the `OrderItem`, rather than using individual parameters like it does now. I thought we could wait until we're sure which data we want to get from the product/variation vs. the order item first.)

## Testing instructions

Prerequisite: the store has [Product Bundles extension](https://woocommerce.com/products/product-bundles/) enabled, and also a bundle product with non-empty items.

No changes on the order form UX are expected:

- Go to the Orders tab
- Tap `+` to create an order
- Tap `Add Products` and select a non-bundle product and a non-bundle product variation
- Select a bundle product from the prerequisite
- Configure the bundle
- Add the products to the order
- On the order form, confirm the product details include a "Bundle" label for the parent bundle product and the variation attributes for any selected product variations. Confirm non-bundle, non-variation products in the order show only the stock status on that line.
- Tap to expand the bundle product and confirm you can re-configure the bundle and your new configuration is saved.


---
